examples/java/native-assets-example/src/main/java/iog/psg/assets/Scenarios.java

```diff
@@ -0,0 +1,107 @@
+        return new StreamObserver<>() {
+            ArrayList<R> buff = new ArrayList();
+
+            @Override
+            public void onNext(E event) {
+                R converted = f.apply(event);
+                System.out.println("Got interim response: " + converted);
+                buff.add(converted);
+            }
+
+            @Override
+            public void onError(Throwable throwable) {
+                System.out.println("Got error: " + throwable.getCause());
+            }
+
+            @Override
+            public void onCompleted() {
+                result.complete(buff.get(buff.size() - 1));
+            }
+        };
+
+    }
+
+    public static void main(String[] args) throws ExecutionException, InterruptedException, CborSerializationException {
+        String apiKey = ""; // apiKey
+        String clientId = ""; // clientId
+        // Create the Api
+        NativeAssetsApi api = NativeAssetsBuilder.create(apiKey, clientId).build();
+        NativeAssetsMultisigApi helper = NativeAssetsBuilder.create(apiKey, clientId).buildMultisig();
+
+        Keys keys = helper.generateKeys();
+        Address targetTransferAddress = helper.generateAddress(keys.getVkey(), Networks.mainnet());
+
+        // create policy with name My Policy and
+        TimeBounds timeBounds = TimeBounds
+                .newBuilder()
+                .setAfter(140215)
+                .setBefore(348215)
+                .build();
+
+        // Create the policy with name `My Policy`
+        CreatePolicyResponse response = api.createPolicy("My Policy", timeBounds)
+                .toCompletableFuture()
+                .get();
+
+        String policyId = response.getPolicy().getPolicyId();
+        Integer depth = 4;
+        CompletableFuture<FundNativeAssetTransaction> fundResult = new CompletableFuture();
+        // Funds the paymentAddress associated with policy with name `policyId`
+        api.fundPolicyAddress(policyId, 10000000l, 4, lastResult(fundResult, r -> r.getFundTx()));
+
+        String assetName = "MyAssetName";
+        NativeAssetId nativeAssetId = NativeAssetId
+                .newBuilder()
+                .setName(assetName)
+                .setPolicyId(policyId)
+                .build();
+        // Create the asset with the policy `policyId` and asset `assetName` on the server
+        CreateNativeAssetResponse assetResponse = api.createAsset(nativeAssetId)
+                .toCompletableFuture()
+                .get();
+
+
+        Map<String, Value> data = new HashMap<>();
+        Struct struct = com.google.protobuf.Struct.newBuilder()
+                .putAllFields(data)
+                .build();
+        //Mint `100` tokens of the native asset defined with `nativeAssetId`
+        CompletableFuture<NativeAssetTransaction> mintResult = new CompletableFuture<>();
+        api.mintAssetWithArbitraryMetadata(nativeAssetId, 100L, null, 4, lastResult(mintResult, r -> r.getAssetTx()));
```

Comment:
TODO replace the null with proper struct, once this fix is in place https://github.com/input-output-hk/psg-services/pull/351